test/base/unit/ControllerSpec.scala

```diff
@@ -107,9 +113,16 @@ trait ControllerSpec
 
   private def getValue(field: Field, cc: AnyRef): String =
     field.get(cc) match {
-      case c: Option[_] if c.isDefined => c.get.toString
-      case c: String                   => c
-      case _                           => ""
+      case c: Option[_] if c.isDefined =>
+        c.get match {
+          case _: Boolean =>
+            throw new RuntimeException(
+              "This test helper function was probably about to bypass fromForm and return 'true' rather than 'yes' and is not able to properly encode boolean forms where the controller expects yes"
+            )
```

Comment:
Better that this fails loudly rather than letting things continue with a potentially incorrect request?

Reply:
I think actually this method is fine, a boolean value as a string is either true or false. If someone now tries to use this properly for a boolean it will thrown an exception needlessly.